ray_on_golem/server/cluster/cluster.py

```diff
@@ -243,3 +266,33 @@ async def _remove_manager_stack(self, stack_key: StackKey) -> None:
             del self._manager_stacks_locks[stack_key]
 
         logger.info(f"Removing stack `%s` done", stack_key)
+
+    def _get_stack_key(self, node_config: NodeConfigData, is_head_node: IsHeadNode) -> StackKey:
+        return (self._get_hash_from_node_config(node_config), is_head_node)
+
+    @staticmethod
+    def _get_hash_from_node_config(node_config: NodeConfigData) -> str:
+        return hashlib.md5(node_config.json().encode()).hexdigest()
+
+    async def _on_node_stop(self, node: ClusterNode) -> None:
```

Comment:
Multiple calls to this function should always be serial. Does it have to be async?

Reply:
Well, there is no need for async, indeed. But as a sync method, it can't be called concurrently, so no locks are required.